pkg/build/registry/buildconfiginstantiate/rest.go

```diff
@@ -178,7 +178,31 @@ func (h *binaryInstantiateHandler) handle(r io.Reader) (runtime.Object, error) {
 	}
 	remaining := h.r.Timeout - time.Now().Sub(start)
 
+	// Attempt to cancel the build if it did not start running
```

Comment:
Hrm.  Certainly ugly to test, but also ugly not to have any sort of certainly this code works.